packages/skills/weather_client_ledger/behaviours.py

```diff
@@ -62,6 +76,63 @@ def act(self) -> None:
                                             protocol_id=OEFMessage.protocol_id,
                                             message=OEFSerializer().encode(search_request))
 
+    def teardown(self) -> None:
+        """
+        Implement the task teardown.
+
+        :return: None
+        """
+        balance = self.context.ledger_apis.token_balance('fetchai', cast(str, self.context.agent_addresses.get('fetchai')))
+        logger.info("[{}]: ending balance on fetchai ledger={}.".format(self.context.agent_name, balance))
+
+
+class MyTransactionBehaviour(Behaviour):
```

Comment:
Currently, transaction message handling from decision maker is implemented as a behaviour. That is conceptually wrong and I would rather implement it as a handler. However, we first need the `Filter` so we can keep a separation between external and internal message filtering. I will work on the `Filter` tomorrow. Then we can make this a `Handler` which listens to `SUPPORTED_PROTOCOL = TransactionMessage.protocol_id`.